drivers/timer/nrf_rtc_timer.c

```diff
@@ -63,7 +61,10 @@ void rtc1_nrf_isr(void *arg)
 	if (!IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
 		u32_t next = last_count + CYC_PER_TICK;
 
-		if (counter_sub(next, t) < MIN_DELAY) {
+		/* As below: we're guaranteed to get an interrupt as
+		 * long as it's set two or more cycles in the future
+		 */
+		if (counter_sub(next, t) < 3) {
```

Comment:
If CYC_PER_TICK = 1 (which is default) , the "next" may be still to close to current counter value for reliable interrupt.